target/arc/semfunc.c

```diff
@@ -121,6 +121,27 @@ arc_gen_add_unsigned_overflow_i64(TCGv_i64 overflow, TCGv_i64 result,
     tcg_temp_free_i64(t1);
 }
 
+void
+arc_gen_rotate_left32_i64(TCGv_i64 destination, TCGv_i64 target_operand,
+                          TCGv_i64 rotate_by)
+{
+    TCGv_i64 temp1 = tcg_temp_new_i64();
+    TCGv_i64 temp2 = tcg_temp_new_i64();
+
+    // Basic shift
+    tcg_gen_shl_i64(temp1, target_operand, rotate_by);
+    // Obtain rotated chunk
+    tcg_gen_subfi_i64(temp2, 32, rotate_by);
+    tcg_gen_shr_i64(temp2, target_operand, temp2);
+    // Assemble
+    tcg_gen_or_i64(destination, temp1, temp2);
```

Comment:
Cannot we use here `tcg_gen_rotl_i32`  or something of a sort?

Reply:
These and the changes for most of the current PRs are for 32 bit operations in 64 bit architecture.
In order to use _i32 instructions I would have to use extraction/extension functions to get 32 bit TCGvs from the 64 bit ones, and vice versa.
I could also use the internal "TCGV_LOW" and "TCGV_HIGH" to "cast" the TCGv but that might be dangerous as they seems purposefully hidden (no other arch uses them, except sparc and only twice).
In order to avoid that extra code, I decide to simply do the logic directly.